pkgs/development/libraries/arpa2common/default.nix

```diff
@@ -0,0 +1,71 @@
+{ lib
+, stdenv
+, fetchFromGitLab
+, cmake
+
+, arpa2cm
+, doxygen
+, e2fsprogs
+, libkrb5
+, libressl
+, libsodium
+, lmdb
+, pkg-config
+, ragel
+}:
+
+stdenv.mkDerivation rec {
+  pname = "arpa2common";
+  version = "2.2.14";
+
+  src = fetchFromGitLab {
+    owner = "arpa2";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-LWsWoHRdLWRSF9JaEwrw+CXm5Azgh7zNeq0a8Z/hijQ=";
+  };
+
+  nativeBuildInputs = [
+    arpa2cm
+    cmake
+    doxygen
+    e2fsprogs
+    libkrb5
+    libressl
+    libsodium
+    lmdb
+    pkg-config
+    ragel
+  ];
+
+  propagatedBuildInputs = [ arpa2cm lmdb libkrb5 libressl ];
+
+  # otherwise the tests fail
+  preConfigure = "mkdir -p build";
+  configurePhase = ''
+    runHook preConfigure
+
+    cd build
+    cmake .. -DCMAKE_INSTALL_PREFIX=$out -DCMAKE_PREFIX_PATH=$out
```

Comment:
Why can't we use the standard cmake phase? Maybe we can change cmakeFlags and preConfigure to use it?

Reply:
If I don't overwrite any phase, the tests fail. If I only include `cmake ..` it's outputting this error.
```nix
       > [ 97%] Built target xs-comm
       > [100%] Built target group-iterate
       > Install the project...
       > -- Install configuration: ""
       > CMake Error at cmake_install.cmake:46 (file):
       >   file cannot create directory: /var/empty/local/lib/pkgconfig.  Maybe need
       >   administrative privileges.
       >
       >
       > make: *** [Makefile:159: install] Error 1
```
I'm not sure how to make it work without overwriting the phases.

Reply:
I think we need to overwrite the directory where it install the pkgconfig file because the above path is not correct.

Reply:
The thing is, there's no pkgconfig file, unless I'm mistaken. Could you point me to some other option?